src/contrasts.jl

```diff
@@ -242,11 +242,13 @@ for contrastType in [:DummyCoding, :EffectsCoding, :HelmertCoding, :SeqDiffCodin
         ## constructor with optional keyword arguments, defaulting to nothing
         $contrastType(; base=nothing, levels::Union{AbstractVector,Nothing}=nothing) = $contrastType(base, levels)
         baselevel(c::$contrastType) = c.base
+        DataAPI.levels(c::$contrastType) = c.levels
     end
 end
 
-# fallback method for other types that might not have base field
+# fallback method for other types that might not have base or level fields
 baselevel(c::AbstractContrasts) = nothing
+DataAPI.levels(c::AbstractContrasts) = nothing
```

Comment:
Is this needed or here just in case? Not a big deal, but it doesn't really follow the DataAPI interface.

Reply:
Well, some kind of "I don't have specified levels" is needed as a fallback, but it could be something else like an empty tuple.

Reply:
OK. I was just wondering whether somebody actually needed to call this on `AbstractContrasts` objects without knowing whether they implement `levels` or not`. Returning an empty tuple would  be problematic as it could also mean "there are no levels".

Reply:
The only place where this is really used is in constructing the `ContrastsMatrix`.  All contrasts EVENTUALLY have to have levels, so whatever sentinel value we use is just to indicate that they haven't been manually specified ahead of time.